src/emqx_misc.erl

```diff
@@ -466,6 +468,63 @@ maybe_mute_rpc_log(Node) ->
             ok
     end.
 
+is_sensitive_key(token) -> true;
+is_sensitive_key("token") -> true;
+is_sensitive_key(<<"token">>) -> true;
+is_sensitive_key(password) -> true;
+is_sensitive_key("password") -> true;
+is_sensitive_key(<<"password">>) -> true;
+is_sensitive_key(secret) -> true;
+is_sensitive_key("secret") -> true;
+is_sensitive_key(<<"secret">>) -> true;
+is_sensitive_key(passcode) -> true;
+is_sensitive_key("passcode") -> true;
+is_sensitive_key(<<"passcode">>) -> true;
+is_sensitive_key(passphrase) -> true;
+is_sensitive_key("passphrase") -> true;
+is_sensitive_key(<<"passphrase">>) -> true;
+is_sensitive_key(_) -> false.
+
+redact(L) when is_list(L) ->
+    lists:map(fun redact/1, L);
+redact(M) when is_map(M) ->
+    maps:map(fun(K, V) ->
+                     redact(K, V)
+             end, M);
+redact(T) when is_tuple(T) ->
+    Redact = case is_nested_tuple(T) of
+                 {true, Elements} ->
+                     redact(Elements);
+                 {false, [Head | Tail]} ->
+                     [Head | redact(Head, Tail)]
+             end,
+    erlang:list_to_tuple(Redact);
+redact(Any) ->
+    Any.
+
+redact(K, V) ->
+    case is_sensitive_key(K) of
+        true ->
+            redact_v(V);
+        false ->
+            redact(V)
+    end.
+
+-define(REDACT_VAL, "*****").
+redact_v(V) when is_binary(V) -> <<?REDACT_VAL>>;
+redact_v(V) when is_list(V) ->
+    case io_lib:printable_list(V) of
+        true ->
+            ?REDACT_VAL;
+        _ ->
+            lists:map(fun redact_v/1, V)
+    end;
```

Comment:
I think it's fare to simply redact the whole value.
unless you have an counter-example ?

Reply:
This is used to solve that can't distinguish list/string when traversing tuple